packages/grid/x-data-grid-pro/src/tests/exportPrint.DataGrid.test.tsx

```diff
@@ -0,0 +1,262 @@
+import * as React from 'react';
+import { DataGrid, GridToolbarExport, GridRowsProp, GridColumns } from '@mui/x-data-grid';
+import { createRenderer, screen, fireEvent } from '@mui/monorepo/test/utils';
+import { expect } from 'chai';
+import { spy } from 'sinon';
+
+describe('<DataGridPro /> - Export - Print', () => {
+  const { render, clock } = createRenderer({ clock: 'fake' });
+
+  const rows: GridRowsProp = [
+    { id: 1, col1: '1-1', col2: '1-2' },
+    { id: 2, col1: '2-1', col2: '2-2' },
+  ];
+  const columns: GridColumns = [{ field: 'col1' }, { field: 'col2' }];
+  const baselineProps = {
+    columns,
+    rows,
+    // A hack to remove the warning on print
+    rowsPerPageOptions: [2, 100],
+  };
+
+  const allBooleanConfigurations = [
+    {
+      printVisible: true,
+      gridVisible: true,
+    },
+    {
+      printVisible: false,
+      gridVisible: true,
+    },
+    {
+      printVisible: true,
+      gridVisible: false,
+    },
+    {
+      printVisible: false,
+      gridVisible: false,
+    },
+  ];
+  // We need `createObjectURL` to test the downloaded value
+  before(function beforeHook() {
+    if (/jsdom/.test(window.navigator.userAgent)) {
+      // Need layouting
+      this.skip();
+    }
+  });
+
+  describe('restore visibility', () => {
+    allBooleanConfigurations.forEach(({ printVisible, gridVisible }) => {
+      it(`should set and restore 'col1' from ${printVisible ? "'visible'" : "'hidden'"} to ${
+        gridVisible ? "'visible'" : "'hidden'"
+      } (visibilityModel)`, async function test() {
```

Comment:
Ok there's too much information in the name of this test case. I would first create two `describe` blocks: one to test the `columnVisibilityModel` prop and another for the `hide` attribute. Then, I would detail better what are the replacements. At first time I had to look at the variable name to know what the first "visible" / "hidden" is referring to. The final test name is not too meaningful.

```
should set and restore 'col1' from 'visible' to 'hidden'
```

Reply:
Now, the message is the following one

```
should have 'currencyPair' 'visible' in print and 'hidden' in screen
```